### ISSUE TITLE
Compiling of deps/phoenix/web/static/js/phoenix.js failed.

### ISSUE THREAD

**Issue by saaji** [phoenixframework/phoenix#1481]
After node modules have been reinstalled from scratch, `mix phoenix.server` started to fail with 

```
Compiling of deps/phoenix/web/static/js/phoenix.js failed. deps/phoenix/web/static/js/phoenix.js: Cannot set property 'Symbol(should not be considered a local binding)' of undefined
```

This issue boils down to the [setTimeout](https://github.com/phoenixframework/phoenix/blob/master/web/static/js/phoenix.js#L722-L729) method in Timer class. Looks like fresh babel 6.4.5 is trying to use `Timer.setTimeout` instead of global `setTimeout` during `this.timer` assignment.

Using `window.setTimeout` explicitly works this issue around:

```js
setTimeout(){
  clearTimeout(this.timer)

  this.timer = window.setTimeout(() => {
    this.tries = this.tries + 1
    this.callback()
  }, this.timerCalc(this.tries + 1))
}
```

**Comment by josevalim** [comment]
Perfect, thank you so much. :) We will work on a fix, we have a new release to do today anyway. /cc @chrismccord

**Comment by chrismccord** [comment]
I'm recreating and fixing now before release

**Comment by chrismccord** [comment]
With bable 6.4.5, I can't recreate this. Can you push your example project to github for me to pull down? Did you generate the app with the phoenix 1.1.2 installer?

**Comment by saaji** [comment]
The project was generated with 1.1.1 I believe. Unfortunately I can not share it. I'll try to put together minimal example.

**Comment by chrismccord** [comment]
We need more info to pin this down, and it's possible this is fixed on Phoenix 1.1.2. Have you upgraded the project to phoenix 1.1.2? Did you follow the upgrade guides?

**Comment by saaji** [comment]
I am not sure if I carefully followed the upgrade guides. It was some time ago.

The issue though can be easily reproduced if you add `deps/phoenix/web/static` to your watched paths in `brunch-config.js`. My watch section looks like:

```
// Phoenix paths configuration
paths: {
  // Dependencies and current project directories to watch
  watched: [
    "deps/phoenix/web/static",
    "web/static",
    "test/static"
  ],

  // Where to compile files to
  public: "priv/static"
},
```

Thank you for quick reaction :) :+1:
